test/e2e/set/scyllacluster/scyllacluster_tls.go

```diff
@@ -244,6 +248,124 @@ var _ = g.Describe("ScyllaCluster", func() {
+		sc, err := f.ScyllaClient().ScyllaV1().ScyllaClusters(f.Namespace()).Create(ctx, sc, metav1.CreateOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		framework.By("Waiting for the ScyllaCluster to rollout (RV=%s)", sc.ResourceVersion)
+		waitCtx1, waitCtx1Cancel := utils.ContextForRollout(ctx, sc)
+		defer waitCtx1Cancel()
+		sc, err = utils.WaitForScyllaClusterState(waitCtx1, f.ScyllaClient().ScyllaV1(), sc.Namespace, sc.Name, utils.WaitForStateOptions{}, utils.IsScyllaClusterRolledOut)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		verifyScyllaCluster(ctx, f.KubeClient(), sc)
+		hosts := getScyllaHostsAndWaitForFullQuorum(ctx, f.KubeClient().CoreV1(), sc)
+		o.Expect(hosts).To(o.HaveLen(1))
+		host := hosts[0]
+
+		servingCABundleConfigMap, err := f.KubeClient().CoreV1().ConfigMaps(f.Namespace()).Get(ctx, fmt.Sprintf("%s-local-serving-ca", sc.Name), metav1.GetOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+		servingCACerts, _ := verification.VerifyAndParseCABundle(servingCABundleConfigMap)
+		o.Expect(servingCACerts).To(o.HaveLen(1))
+
+		adminClientSecret, err := f.KubeClient().CoreV1().Secrets(f.Namespace()).Get(ctx, fmt.Sprintf("%s-local-user-admin", sc.Name), metav1.GetOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+		_, adminClientCertBytes, _, adminClientKeyBytes := verification.VerifyAndParseTLSCert(adminClientSecret, verification.TLSCertOptions{
+			IsCA:     pointer.Ptr(false),
+			KeyUsage: pointer.Ptr(x509.KeyUsageKeyEncipherment | x509.KeyUsageDigitalSignature),
+		})
+
+		servingCAPool := x509.NewCertPool()
+		servingCAPool.AddCert(servingCACerts[0])
+		adminTLSCert, err := tls.X509KeyPair(adminClientCertBytes, adminClientKeyBytes)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		const loaders = 3
+
+		for i := 0; i < loaders; i++ {
+			clusterConfig := gocql.NewCluster(hosts...)
+			clusterConfig.Port = 9142
+			clusterConfig.SslOpts = &gocql.SslOptions{
+				Config: &tls.Config{
+					ServerName:         host,
+					InsecureSkipVerify: false,
+					Certificates:       []tls.Certificate{adminTLSCert},
+					RootCAs:            servingCAPool,
+				},
+			}
+
+			di := insertAndVerifyCQLData(ctx, hosts, utils.WithClusterConfig(clusterConfig))
+			defer di.Close()
+
+			trafficCtx, trafficCtxCancel := context.WithCancel(ctx)
+			defer trafficCtxCancel()
+
+			stopTraffic, err := di.StartContinuousReads(trafficCtx)
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			defer func() {
+				err := stopTraffic()
+				o.Expect(err).NotTo(o.HaveOccurred())
+			}()
+		}
+
+		framework.By("Adding new DNS domain to rotate certificates")
+		testDNSDomain := fmt.Sprintf("%s.scylla-operator.scylladb.com", rand.String(6))
+
+		sc.ManagedFields = nil
+		sc.ResourceVersion = ""
+		sc.Spec.DNSDomains = append(sc.Spec.DNSDomains, testDNSDomain)
+		scData, err := runtime.Encode(scheme.Codecs.LegacyCodec(scyllav1.GroupVersion), sc)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		// TODO: Use generated Apply method when our clients have it.
```

Comment:
Shall we have an issue for it? I feel like these TODOs get lost in the void otherwise

Reply:
done

Reply:
Let's link the issue in this comment as well